### PR TITLE
Don't handle symbolic sets in `Kore.Builtin.Set.evalDifference`

### LINKED ISSUE #2700: Set simplification not working
On K master commit (72d33ae0e0), the following very simple spec does not work. This affects some KEVM proofs which are being ported.

```
requires "domains.md"

module TEST
    imports INT
    imports SET

    configuration <k> $PGM:Step </k>

    syntax Step ::= run ( Set ) | done ( Set )

    rule <k> run(S) => done(S) ... </k>

    syntax Bool ::= pred ( Int ) [function, no-evaluators]

    rule (SetItem(X) REST) -Set SetItem(Y) => SetItem(X) (REST -Set SetItem(Y)) requires pred(X) andBool notBool pred(Y) [simplification]

    rule pred(X) => false requires 1 <=Int X andBool X <=Int 9 [simplification]
endmodule

module TEST-SPEC
    imports TEST

    claim <k> run(SetItem(1) -Set SetItem(X)) => done(SetItem(1)) ... </k> requires pred(X)

endmodule
```

To reproduce the failure:

```
kompile --backend haskell test.k --main-module TEST --syntax-module TEST
kprove test.k --def-module TEST --spec-module TEST-SPEC
```

### PR DESCRIPTION
**Pull request by geo2a**
* `evalDifference` now only handles concrete sets, offloading the
   symbolic case to the user-defined rules
*  Remove unit tests for the symbolic arguments of `-Set`.

Partially fixes #2700

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

**Comment by ana-pantilie** [comment]
Let's run the tests from https://github.com/kframework/k/tree/master/k-distribution/tests with this PR as well.

File: test/issue-2700/set-difference-spec.k [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+          <k> run(SetItem(1) -Set SetItem(10)) => done(SetItem(1)) ... </k>
+
+    claim [claim.diff.left.concrete.singleton]:
+          <k> run(SetItem(1) -Set SetItem(X)) => done(SetItem(1)) ... </k>
+      requires pred(X)
+
+    // This claim does not pass, am I missing something?
+    // claim [claim.diff.left.symbolic.singleton]:
+    //       <k> run(SetItem(X) -Set SetItem(10)) => done(SetItem(X)) ... </k>
+    //   requires notBool pred(X)
```
Comment (ana-pantilie) [inline test/issue-2700/set-difference-spec.k:26]: For the `concat` equation to apply, it requires `pred(10)` to be `true`. The `pred.false` equation doesn't apply for `pred(10)`, so it remains unevaluated.

I think adding `pred(10)` in the claim's requires should work, or adding another equation for `pred` which handles the other cases (or specifically the case for `10`).

Comment (geo2a) [reply test/issue-2700/set-difference-spec.k:26]: The LHS of this claim is not matched even if I remove `pred` altogether. It appears that this needs more investigation, I'll add some traces again tomorrow and see what I could find.

Comment (geo2a) [reply test/issue-2700/set-difference-spec.k:26]: @ana-pantilie I have added an even more reduced example to demonstrate the issue. Apparently a symbolic variable is not matched with a concrete value when it should. Here are the terms that I think should be matched, produced with a `trace` in `attemptEquation`, like we did before:
```
term1
/* Fl Fn D Spa */
LblSet'Coln'difference{}(
    /* Fl Fn D Spa */
    /* InternalSet: */ /* element: */ LblSetItem{}(
        /* Fl Fn D Spa */
        /* Inj: */ inj{SortInt{}, SortKItem{}}(
            /* Fl Fn D Sfa */ EquationVarX:SortInt{}
        )
    ),
    /* Fl Fn D Spa */
    /* InternalSet: */ /* element: */ LblSetItem{}(
        /* Fl Fn D Spa */
        /* Inj: */ inj{SortInt{}, SortKItem{}}(
            /* Fl Fn D Sfa */ EquationVarY:SortInt{}
        )
    )
)
term2
/* Fl Fn D Spa */
LblSet'Coln'difference{}(
    /* Fl Fn D Sfa */
    /* InternalSet: */ /* element: */ LblSetItem{}(
        /* Fl Fn D Sfa */
        /* Inj: */ inj{SortInt{}, SortKItem{}}(
            /* Fl Fn D Sfa */ ConfigVarX:SortInt{}
        )
    ),
    /* Fl Fn D Sfa Cl */
    /* InternalSet: */ /* concrete element: */ LblSetItem{}(
        /* Inj: */ inj{SortInt{}, SortKItem{}}(\dv{SortInt{}}("10"))
    )
)
```

Comment (ana-pantilie) [reply test/issue-2700/set-difference-spec.k:26]: I don't think it should be handled by that case, could you add a unit test for this similar to the ones I added in the new matching case PR?

Comment (geo2a) [reply test/issue-2700/set-difference-spec.k:26]: Sure, will do.

By the way, if I change `10` to `Y`, the claim passes, i.e. the "structurally equal" works for variables.

Comment (geo2a) [reply test/issue-2700/set-difference-spec.k:26]: I added a [unit tests](https://github.com/kframework/kore/pull/2737/commits/2f5970aa138f33f7015b74135ce81517d105cd1d) that I think should correspond to the issue that arises in the [example](https://github.com/kframework/kore/pull/2737/commits/89265c1977cd172cee0bdf48a9ae8021e885cc9b).

**Pull request closed without merging**